### PR TITLE
`ultralytics 8.0.57` Comet, AMP, Classify, Docker updates

### LINKED ISSUE #1148: YOLOV8 GPU train all loss is NAN  and P/R/map is 0
### Search before asking

- [ ] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

all loss is NAN and P/R/map is 0 when the user-defined data set GPU is trained! CUDA Change from 11.7 to 11.6 still can't train properly.Using the CPU works fine, but is often too long。Is there any solution？

1. **Environment**

- yolov8 install：
> pip install ultralytics
> pip install onnx
> pip install onnxruntime-gpu
>conda install pytorch torchvision torchaudio pytorch-cuda=11.6 -c pytorch -c nvidia

- Ultralytics YOLOv8.0.45   Python-3.9.16  torch-1.13.1+cu116 CUDA:0 (NVIDIA GeForce GTX 1660Ti, 6144MiB)
- Windows10 CPU:i7-10750H, 32 GB RAM
- command:yolo task=detect mode=train model=./datasets/yolov8s.yaml data=./datasets/dataset.yaml epochs=100  batch=10 imgsz=640 resume=True workers=4

2. **Datasets**
- Data sources: Open Images V6
- Dataset structure：
dataset_struck
image
image
image
image

3. **Result**

result
result2

### Additional

_No response_

### LINKED ISSUE #1172: Training loses are nan with GPU, but with CPU it trains fine (with C3TR block)
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training

### Bug

I am training [yolov8n-seg.yaml](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/models/v8/seg/yolov8n-seg.yaml) backbone but with a C3TR block in line 8 with my custom dataset. My custom dataset is fine, no corrupted labels or anything. When training with the yolov8n.yaml, everything works fine with GPU training. When training with transformer block C3TR and device='cpu', everything trains fine too. But including the C3TR module doesn't work when using GPU for training. I tried using coco128.yaml, different workers number, lower lr0, higher batch. To so sum up, I tried everything I found as an answer in other similar issues. I also tried v5loader=True. Nothing works. So my guess is that it's related to the GPU/CUDA....

### Environment

Ultralytics YOLOv8.0.46 🚀 Python-3.10.9 torch-1.13.1+cu117

My Ultralytics YOLO version is 8.0.46. I also tried from versions 8.0.41 up to 45 and the same happens.

My GPU is a GV100 My CUDA is 11.7. I also tried with CUDA 11.3. 

My Python is 3.10. I also tried Python 3.7, 3.8 and 3.9.

Pytorch 1.13.1, also tried Pytorch 1.12.1

Ubuntu 20.04

### Minimal Reproducible Example

backbone:
  - [-1, 1, Conv, [64, 3, 2]]  # 0-P1/2
  - [-1, 1, Conv, [128, 3, 2]]  # 1-P2/4
  - [-1, 3, C2f, [128, True]]
  - [-1, 1, Conv, [256, 3, 2]]  # 3-P3/8
  - [-1, 6, C2f, [256, True]]
  - [-1, 1, Conv, [512, 3, 2]]  # 5-P4/16
  - [-1, 6, C2f, [512, True]]
  - [-1, 1, Conv, [1024, 3, 2]]  # 7-P5/32
  - [-1, 3, C3TR [1024, False]] #<-------------------change introduced by me, from - [-1, 3, C2f [1024, True]]
  - [-1, 1, SPPF, [1024, 5]]  # 9

head:
  - [-1, 1, nn.Upsample, [None, 2, 'nearest']]
  - [[-1, 6], 1, Concat, [1]]  # cat backbone P4
  - [-1, 3, C2f, [512]]  # 13
  - [-1, 1, nn.Upsample, [None, 2, 'nearest']]
  - [[-1, 4], 1, Concat, [1]]  # cat backbone P3
  - [-1, 3, C2f, [256]]  # 17 (P3/8-small)
  - [-1, 1, Conv, [256, 3, 2]]
  - [[-1, 12], 1, Concat, [1]]  # cat head P4
  - [-1, 3, C2f, [512]]  # 20 (P4/16-medium)
  - [-1, 1, Conv, [512, 3, 2]]
  - [[-1, 9], 1, Concat, [1]]  # cat head P5
  - [-1, 3, C2f, [1024]]  # 23 (P5/32-large)
  - [[15, 18, 21], 1, Segment, [nc, 32, 256]]  # Detect(P3, P4, P5)

Code for training:

`model.train(data="dataset.yaml", epochs=100, imgsz=320, name='yolov8n-seg-tr', project='transformer_block', save_period=1)`

### Additional

_No response_

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by glenn-jocher**
Usage:

```
yolo train amp=False
```

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Enhancements to Ultralytics YOLOv8 Docker configuration, training options, and task support.

### 📊 Key Changes
- Updated the Docker base image to a specific PyTorch image with CUDA support.
- Added Automatic Mixed Precision (AMP) training configuration in training documentation and code.
- Expanded YOLO tasks to include `pose` estimation along with `detect`, `segment`, and `classify`.
- Introduced new YOLO modes: `track` for object tracking and `benchmark` for performance testing.
- Updated the tutorial notebook's unzip command for dataset preparation.
- Bumped up version to 8.0.57 to reflect these updates.
- Improved checks and fallbacks for dataset configurations missing 'names' or 'nc' (class count).

### 🎯 Purpose & Impact
- Docker image change ensures that users work with an optimized environment for their GPU, potentially improving performance. 🚀
- AMP training can significantly speed up computation on compatible hardware with reduced precision operations, making training more efficient. ⚙️
- New `pose` task and `track`, `benchmark` modes expand the range of applications YOLOv8 can address, increasing the toolkit's versatility. 🧍‍♀️🔍
- The tutorial improvements and version bump make starting out with YOLO more straightforward for newcomers and signal ongoing development. 🎉
- Fixes in dataset handling prevent potential errors due to configuration oversights, simplifying the data setup process for users. 🛠️

### CODE REVIEW

**Comment by JianDan0212** [comment]
Using this can indeed solve the problem of nan, but the parameters (P, R, mAP50, mAP50-95) are all 0, and the loss value of the training process has not converged.

**Comment by glenn-jocher** [comment]
@JianDan0212 if you can train COCO128 correctly then your dataset is the cause of the problems.
